src/servers/cron.ts

```diff
@@ -17,43 +18,62 @@ import { SpecterWallet } from "@core/specter-wallet"
 import { activateLndHealthCheck } from "@services/lnd/health"
 
 const main = async () => {
-  const mongoose = await setupMongoConnection()
+  const results: Array<boolean> = []
+  await asyncRunInSpan(
+    "cron.main",
+    {
+      [SemanticAttributes.CODE_FUNCTION]: "main",
+      [SemanticAttributes.CODE_NAMESPACE]: "cron",
+    },
+    async () => {
+      const mongoose = await setupMongoConnection()
 
-  const rebalance = () => {
-    const specterWalletConfig = getSpecterWalletConfig()
-    const specterWallet = new SpecterWallet({
-      logger: baseLogger,
-      config: specterWalletConfig,
-    })
-    return specterWallet.tentativelyRebalance()
-  }
+      const rebalance = () => {
+        const specterWalletConfig = getSpecterWalletConfig()
+        const specterWallet = new SpecterWallet({
+          logger: baseLogger,
+          config: specterWalletConfig,
+        })
+        return specterWallet.tentativelyRebalance()
+      }
 
-  const updatePendingOnChainPayments = () =>
-    updateUsersPendingPayment({ onchainOnly: true })
+      const updatePendingOnChainPayments = () =>
+        updateUsersPendingPayment({ onchainOnly: true })
 
-  const tasks = [
-    updateEscrows,
-    updatePendingLightningTransactions,
-    deleteExpiredInvoiceUser,
-    deleteFailedPaymentsAttemptAllLnds,
-    rebalance,
-    updateRoutingFees,
-    updatePendingOnChainPayments,
-  ]
+      const tasks = [
+        updateEscrows,
+        updatePendingLightningTransactions,
+        deleteExpiredInvoiceUser,
+        deleteFailedPaymentsAttemptAllLnds,
+        rebalance,
+        updateRoutingFees,
+        updatePendingOnChainPayments,
+      ]
 
-  const results: Array<boolean> = []
-  for (const task of tasks) {
-    try {
-      baseLogger.info(`starting ${task.name}`)
-      await task()
-      results.push(true)
-    } catch (error) {
-      baseLogger.error({ error }, `issue with task ${task.name}`)
-      results.push(false)
-    }
-  }
+      for (const task of tasks) {
+        await asyncRunInSpan(
+          `cron.${task.name}`,
+          {
+            [SemanticAttributes.CODE_FUNCTION]: task.name,
+            [SemanticAttributes.CODE_NAMESPACE]: "cron",
+          },
+          async () => {
+            try {
+              baseLogger.info(`starting ${task.name}`)
+              await task()
+              results.push(true)
+            } catch (error) {
+              baseLogger.error({ error }, `issue with task ${task.name}`)
+              results.push(false)
+              return error
```

Comment:
Do we need this return if the containing `await asyncRunInSpan` isn't being assigned?

Reply:
Its so that `asyncRunInSpan` will auto add the error via `span.recordException`